gui/api/resources.py

```diff
@@ -3535,7 +3568,8 @@ def dehydrate(self, bundle):
                         'id': bundle.obj.id
                     }
                 )
-        except Exception:
+        except Exception as err:
+            bundle.data['cert_DN'] = "ERROR: " + str(err)
             # There was an error parsing this Certificate object
             # Creating a sentinel file for the alertmod to pick it up
             with open('/tmp/alert_invalidcert_{0}'.format(bundle.obj.cert_name),
```

Comment:
While we are here. Do you think we can switch from sentinel files to new one-shot alert system?

Example is 
```
class VMWareLoginFailedAlertSource(OneShotAlertSource):
    level = AlertLevel.WARNING
    title = "VMWare login failed"

    def create(self, args):
        return Alert("VMWare login to %(hostname)s failed: %(error)s", args)

    def delete(self, alerts, query):
        hostname = query

        return list(filter(
            lambda alert: alert.args["hostname"] != hostname,
            alerts
        ))
```

```
        self.middleware.call_sync("alert.oneshot_create", "VMWareLoginFailed", {
            "hostname": vmsnapobj["hostname"],
            "error": vmlogin_fail,
        })
```

```
self.middleware.call_sync("alert.oneshot_delete", "VMWareLoginFailed", vmsnapobj["hostname"])
```

Reply:
Sure @themylogin we can. However, do you think it would be better if I took care of this in crypto plugin ? The exception clause here is very unlikely to be triggered - or do you suggest we do it in both places ?

Reply:
Both is a bad idea as it would result in duplicate alerts if I am not wrong 💭

Reply:
No, it won't result in duplicate alerts, so we should keep it in both places